Microsoft.Toolkit.Uwp.UI.Controls.Core/ImageEx/ImageExBase.Source.cs

```diff
@@ -165,64 +160,96 @@ private async Task LoadImageAsync(Uri imageUri)
+        /// This method is provided in case a developer would like their own custom caching strategy for <see cref="ImageExBase"/>.
+        /// By default it uses the built-in UWP cache provided by <see cref="BitmapImage"/> and
+        /// the <see cref="Image"/> control itself. This method should call <see cref="AttachSource(ImageSource)"/>
+        /// to set the retrieved cache value to the image. <see cref="CurrentSourceUri"/> may be checked
+        /// after retrieving a cached image to ensure that the current resource requested matches the one
+        /// requested by the <see cref="AttachCachedResourceAsync(Uri)"/> parameter.
+        /// <see cref="OnNewSourceRequested(object)"/> may be used in order to signal any cancellation events
+        /// using a <see cref="CancellationToken"/> to the call to the cache, for instance like the Toolkit's
+        /// own <see cref="CacheBase{T}.GetFromCacheAsync(Uri, bool, CancellationToken, List{KeyValuePair{string, object}})"/> in <see cref="ImageCache"/>.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// try
+        /// {
+        ///     var propValues = new List&lt;KeyValuePair&lt;string, object>>();
+        ///
+        ///     if (DecodePixelHeight > 0)
+        ///     {
+        ///         propValues.Add(new KeyValuePair&lt;string, object>(nameof(DecodePixelHeight), D ecodePixelHeight));
+        ///     }
+        ///     if (DecodePixelWidth > 0)
+        ///     {
+        ///         propValues.Add(new KeyValuePair&lt;string, object>(nameof(DecodePixelWidth), D ecodePixelWidth));
+        ///     }
+        ///     if (propValues.Count > 0)
+        ///     {
+        ///         propValues.Add(new KeyValuePair&lt;string, object>(nameof(DecodePixelType), DecodePixelType));
+        ///     }
+        ///
+        ///     // A token could be provided here as well to cancel the request to the cache,
+        ///     // if a new image is requested. That token can be canceled in the OnNewSourceRequested method.
+        ///     var img = await ImageCache.Instance.GetFromCacheAsync(imageUri, true, initializerKeyValues: propValues);
+        ///
+        ///     lock (LockObj)
+        ///     {
+        ///         // If you have many imageEx in a virtualized ListView for instance
+        ///         // controls will be recycled and the uri will change while waiting for the previous one to load
+        ///         if (CurrentSourceUri == imageUri)
+        ///         {
+        ///             AttachSource(img);
+        ///             ImageExOpened?.Invoke(this, new ImageExOpenedEventArgs());
+        ///             VisualStateManager.GoToState(this, LoadedState, true);
+        ///         }
+        ///     }
+        /// }
+        /// catch (OperationCanceledException)
+        /// {
+        ///     // nothing to do as cancellation has been requested.
+        /// }
+        /// catch (Exception e)
+        /// {
+        ///     lock (LockObj)
```

Comment:
Just a small note to remind to update this XML docs and remove the `try/catch` logic if we do decide to go with the previous idea and to move this control flow logic up one scope, since consumers here wouldn't need to repeat this anymore at that point. 🙂

Reply:
Updated!